src/utils/api.js

```diff
@@ -0,0 +1,42 @@
+export const baseUrl = "http://localhost:3001";
+
+export const checkResponse = (res) => {
+  if (res.ok) {
+    return res.json();
+  }
+  return Promise.reject(`Error: ${res.status}`);
+};
+
+export const handleRequest = (url, options = {}) => {
+  return fetch(url, options).then(checkResponse);
+};
+
+export const signIn = (email, password, username) => {
+  return fetch(`${baseUrl}/signin`, {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({ email, password, username }),
+  }).then(checkResponse);
+};
```

Comment:
You can use `handleRequest()` everywhere, to avoid `.then(checkResponse);` duplication


```js
export const handleRequest = (url, options = {}) => {
  return fetch(url, options).then(checkResponse);
};

export const signIn = (email, password, username) => {
  return handleRequest(`${baseUrl}/signin`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ email, password, username }),
  }) 
};

export const signUp = (email, password, username) => {
  return handleRequest(`${baseUrl}/signup`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({ email, password, username }),
  }) 
};
```